### PR TITLE
Reorganize Food Recipes, "Balance" Tweaks

### PR DESCRIPTION
**Pull request by Owai-Seek**
-Reorganized -ALL- foods in the crafting menu.
-Made a section for Eastern Dishes, Mexican Dishes, and Sweets
-Made Bowls, Salads, Soups, Eggs Benedict, and Omelettes normal sized.
-Moved Easter Recipes to the food files, fixing it in the process.
-Capitalized a few things.

## About The Pull Request

Made subsections in recipes files to organize things based on type, as well as alphabetical sorting. Some minor spellchecks and "balance" changes.

Section Examples:
Bread, Toast, and Misc Bread.
Standard, Exotic, Mystical, and Colored burgers.
Fruit Cake, Fancy Cake, Special Cake
Sorted Frozen Section
Kebabs, Spider, Fish, and Misc (in Meat)
A new Mexican file for Burritos, Tacos, Enchiladas, ect.
A thinned out Misc section.
An only -slightly- organised Pastry section. (Donuts are now sorted.)
Fruit, Other Pies, and Tarts.
Alphabetized Pizza, as well as a separation of Arnold/Anomaly from regular pies.
Sorted Salads
Sandwiches -mostly- untouched.
Soups and Spaggetti Sorted
Sushi is now Eastern
New Sweets Category

## Why It's Good For The Game

Foods are an absolute mess to find. There was a massive unorganized cluster of related foods in the misc category, there was toast in both breads /and/ toast, things werent put together in any cohesive way at all, and it made me irritated.

## Changelog
:cl:
tweak: Reorganized all food recipes, (hopefully) making things easier to find.
balance: Trays now have a whitelist, allowing them to pick up normal sized foods, bowls, glassware, booze, ect.
fix: Easter foods are no longer their own Misc Food Category on the top of the menu.
spellcheck: fixed a few typos/capitalization consistency.
tweak: Mexican Foods are their own Subcategory.
tweak: Donuts are their own Subcategory
tweak: Moved Sweets from Misc Food in with Pies. Renamed to Pies & Sweets
/:cl:

## (Original) Screenshots

kwd1I2IQuS
TTMbVo9ag3
UcXs8XB41j
CMDUhLjTfb

## Category Updates

fJbidtEPj9
wMWuUF4XMp
MW6UZaLMnP
x5oDdwZJ2Y
1MSyMsgvSu
image

### CODE REVIEW

**Comment by CameronWoof** [comment]
Screenshots?

**Comment by CameronWoof** [comment]
Looks very nice. I especially like the cultural sections.

**Comment by timothyteakettle** [comment]
you put toast under bread and for this I can never forgive you

File: code/__DEFINES/construction.dm [hunk 1]
```diff
@@ -102,7 +104,7 @@
 #define CAT_SANDWICH	"Sandwiches"
 #define CAT_SOUP	"Soups"
 #define CAT_SPAGHETTI	"Spaghettis"
-#define CAT_FISH	"Fish"
```
Comment (Trilbyspaceclone) [inline code/__DEFINES/construction.dm:105]: Did you mean to replace Fish?

File: code/__DEFINES/construction.dm [hunk 2]
```diff
@@ -102,7 +104,7 @@
...
+#define CAT_SWEETS	"Sweets"
```
Comment (Ghommie) [inline code/__DEFINES/construction.dm]: Being a super-category of most cakes, pies and pastries, it is likely to cause some confusion (specially with the latter).
Just rename the "Pastries" category to "Confectioneries".

File: code/modules/food_and_drinks/food/customizables.dm [hunk 3]
```diff
@@ -294,7 +294,7 @@
 	icon_state = "bowl"
 	reagent_flags = OPENCONTAINER
 	materials = list(MAT_GLASS = 500)
-	w_class = WEIGHT_CLASS_NORMAL
+	w_class = WEIGHT_CLASS_SMALL
```
Comment (Ghommie) [inline code/modules/food_and_drinks/food/customizables.dm]: A better long term solution would be adding a "size whitelist" variable to the storage component for typecaches of items that can be inserted regardless of size and then whitelist bowls and food reagent containers for the tray.

File: code/modules/food_and_drinks/food/snacks_other.dm [hunk 4]
```diff
@@ -674,3 +674,49 @@
...
+	icon_state = "briochecake"
+	slice_path = /obj/item/reagent_containers/food/snacks/cakeslice/brioche
+	slices_num = 6
+	bonus_reagents = list(/datum/reagent/consumable/nutriment = 10, /datum/reagent/consumable/nutriment/vitamin = 2)
+
+/obj/item/reagent_containers/food/snacks/cakeslice/brioche
+	name = "brioche cake slice"
+	desc = "Delicious sweet-bread. Who needs anything else?"
+	icon_state = "briochecake_slice"
+	filling_color = "#FFD700"
```
Comment (Ghommie) [inline code/modules/food_and_drinks/food/snacks_other.dm]: Being a cake subtype, this is better found in the cakes file.

File: code/modules/food_and_drinks/food/snacks_other.dm [hunk 5]
```diff
@@ -674,3 +674,49 @@
...
+
+/obj/item/reagent_containers/food/snacks/scotchegg
+	name = "scotch egg"
+	desc = "A boiled egg wrapped in a delicious, seasoned meatball."
+	icon_state = "scotchegg"
+	bonus_reagents = list(/datum/reagent/consumable/nutriment = 2, /datum/reagent/consumable/nutriment/vitamin = 2)
+	bitesize = 3
+	filling_color = "#FFFFF0"
+	list_reagents = list(/datum/reagent/consumable/nutriment = 6)
```
Comment (Ghommie) [inline code/modules/food_and_drinks/food/snacks_other.dm]: Egg snacks file.

File: code/modules/food_and_drinks/recipes/tablecraft/recipies_sweets.dm [hunk 6]
```diff
@@ -0,0 +1,117 @@
+// see code/module/crafting/table.dm
```
Comment (Ghommie) [inline code/modules/food_and_drinks/recipes/tablecraft/recipies_sweets.dm]: File name typo. Also this should be merged with the pastries file.

**Review by Ghommie — CHANGES_REQUESTED** [review]
Most times, food recipes are better indexed by typology, main ingredients and food group rather than cultural origins to help people choose what they want to cook based on the available ingredients.

So no, I can't approve of the mexican and eastern categories and you should revert them.

**Comment by CameronWoof** [comment]
The culture categories are useful for putting together a themed menu, which I think is valuable. 

Also, perhaps a "seafood" category would be helpful.

File: code/__DEFINES/construction.dm [hunk 7]
```diff
@@ -92,17 +92,19 @@
 #define CAT_BREAD	"Breads"
 #define CAT_BURGER	"Burgers"
 #define CAT_CAKE	"Cakes"
+#define CAT_DONUT	"Donuts"
+#define CAT_EASTERN	"Eastern & Fish"
```
Comment (Ghommie) [inline code/__DEFINES/construction.dm]: I told you not to fill this category with miscellaneous eastern food. Bring those recipes where you found em.

File: code/modules/food_and_drinks/food/snacks_egg.dm [hunk 8]
```diff
@@ -124,7 +124,6 @@
 	bonus_reagents = list(/datum/reagent/consumable/nutriment = 1, /datum/reagent/consumable/nutriment/vitamin = 2)
 	list_reagents = list(/datum/reagent/consumable/nutriment = 8, /datum/reagent/consumable/nutriment/vitamin = 1)
 	bitesize = 1
-	w_class = WEIGHT_CLASS_NORMAL
```
Comment (Ghommie) [inline code/modules/food_and_drinks/food/snacks_egg.dm]: Don't remove this.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Comment by Ghommie** [comment]
And yea, the "Seafood" category would be nice.

File: code/modules/food_and_drinks/food/snacks_egg.dm (hunk 8 above)
Comment (Owai-Seek) [reply code/modules/food_and_drinks/food/snacks_egg.dm]: Forgot to restore that with the whitelisting.

**Comment by Owai-Seek** [comment]
Don't merge this just yet, I'm going to make Fish into Seafood for when crab stuff gets ported.

**Review by Ghommie — APPROVED** [review]

**Pull request merged**